program_map.go

```diff
@@ -1,54 +1,41 @@
 package astits
 
-import "sync"
-
 // programMap represents a program ids map
 type programMap struct {
-	m *sync.RWMutex
-	p map[uint16]uint16 // map[ProgramMapID]ProgramNumber
+	p map[uint32]uint16 // map[ProgramMapID]ProgramNumber
```

Comment:
What is the metrics difference made by switching from uint32 to uint16 here?

Reply:
[Benchmark](https://gist.github.com/k-danil/37b52bc9dd9f2e8c9c92119a6aa6c514) gave me following results:
```
BenchmarkMaps
BenchmarkMaps/uint8
BenchmarkMaps/uint8-8         	 2317094	       505.0 ns/op	       0 B/op	       0 allocs/op
BenchmarkMaps/uint16
BenchmarkMaps/uint16-8        	 2190326	       548.1 ns/op	       0 B/op	       0 allocs/op
BenchmarkMaps/uint32
BenchmarkMaps/uint32-8        	 3035148	       395.3 ns/op	       0 B/op	       0 allocs/op
BenchmarkMaps/uint64
BenchmarkMaps/uint64-8        	 3026098	       396.4 ns/op	       0 B/op	       0 allocs/op
PASS
```
This will be absolutely neglectable, but every packet should be passed to packetPool and then checked via programMap in packetAccumulator.add().
Profiling data is kind of unstable (600sec run on 10mbps stream): 
- packetPool: before 90-120ms after 30-50ms
- programMap: befor 20-30ms after unmeasurable